docs/examples/sensor-data.md

```diff
@@ -0,0 +1,115 @@
+Analyzing Sensor Data With Kafka and DynamoDB
+================================================
+
+In this example, we will walk through processing sensor data from IOT devices using Bytewax.
+
+In this scenario, IOT devices are streaming real-time data that is being written to Kafka. Our goals are to:
+* Read data from Kafka
+* Build up a window of events from each sensor, based on it's id
+* Write those updated windows to DynamoDB every 5 seconds
+
+Let's start by setting up Bytewax to read from Kafka. We'll begin by importing some dependencies and instantiating our Bytewax Dataflow.
+
+``` python
+import json
+
+from bytewax.dataflow import Dataflow
+from bytewax.inputs import KafkaInputConfig
+
+flow = Dataflow()
+flow.input("inp", KafkaInputConfig(brokers=["localhost:9092"], topic="sensor_topic"))
+```
+
+Our sensor payload is a JSON encoded string in Kafka. Let's write a function that deserializes data from Kafka, and returns it as a dictionary. In addition to the payload, we are also returning the `id` of the sensor that captured this data. Bytewax will use this key to route all of the data for a sensor to the same worker.
+
+``` python
+def deserialize(key_bytes__payload_bytes):
+    _key_bytes, payload_bytes = key_bytes__payload_bytes
+    payload = json.loads(payload_bytes)
+    return payload["id"], payload
+
+flow.map(deserialize)
+```
+
+The goal of our example Dataflow is to accumulate sensor data over a defined period of time to produce an average reading. To do that, we'll use one of Bytewax's stateful windowing fold_window`.
+
+In this example, we want our sensor readings to be grouped together based on the timestamp that the sensor recorded in the data when the data was produced. To do that, we need to define a function that will read that time from the sensor event and convert it to UTC.
+
+``` python
+from datetime import datetime
+
+def get_event_time(event):
+    return datetime.fromisoformat(event["time"])
+```
+
+There are two important considerations when using event times to construct windows. The first is that we can't guarantee that sensor events will be read from Kafka in order, and the second is that we can't guarantee that we will have received all of the events that we should consider in a timely fashion. Some sensors could lose their connection to the Internet and send data well after it was recorded. In this way, event time processing is a tradeoff between latency and correctness.
```

Comment:
```suggestion
There are two important considerations when using event times to construct windows:

 Event Orderliness: We cannot guarantee that sensor events will be read from Kafka in order, so...
 
 Event Lateness: We can't guarantee that we will have received all of the events that we should consider in a timely fashion, since sensors could lose their connection to the Internet and send data well after it was recorded. 
 
 In this way, event time processing is a tradeoff between latency and correctness.
```

Optionally we incorporate how we address ^^ in here